src/lib/cache/context.go

```diff
@@ -0,0 +1,35 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"context"
+)
+
+type cacheKey struct{}
+
+// FromContext returns cache from context
+func FromContext(ctx context.Context) (Cache, bool) {
+	c, ok := ctx.Value(cacheKey{}).(Cache)
+	return c, ok
+}
+
+// NewContext returns new context with cache
```

Comment:
Why config manager needs context?

Reply:
Config manager does not need this, this is preparing for the DAO layers.